*: make go/v3 the default plugin

estroz opened the pull request:
**Description of the change:**
- hack/generate/samples: update webhook marker and a few other nits
- internal/cmd/operator-sdk/cli: make go/v3 the default

**Motivation for the change:** now that go/v3 is stable, it can be the default plugin. This is not a breaking change because go/v2 is still supported.


**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

estroz commented:
Waiting on https://github.com/kubernetes-sigs/kubebuilder/pull/1882

/hold

estroz commented:
/hold cancel

estroz commented:
Blocked by #4315

camilamacedo86 commented:
HHi @estroz, 

It fails  with the error: 

```
warning: error when running kustomize build: kustomize build config/default failed with error: (exec: "kustomize": executable file not found in $PATH) 
```

So, we might need to check if we have not an issue with the makefile target. 

https://github.com/operator-framework/operator-sdk/pull/4307/checks?check_run_id=1553378087#step:5:427

camilamacedo86 commented on changelog/fragments/go-v3-default.yaml:
```diff
@@ -0,0 +1,28 @@
+        incompatible with projects created by go/v2. You are not required to upgrade
+        and your go/v2 project will continue to work with new operator-sdk versions.
+        If you wish to upgrade, check out the upstream
+        [migration guide](https://master.book.kubebuilder.io/migration/plugin/plugins.html).
```
IMO we need to add here to update the layout from `layout: go.kubebuilder.io/v2` to `layout: go.kubebuilder.io/v3-alpha`
Then, I think would be better the kb link at the end for users are able to look for further info.

estroz replied:
I'd rather defer completely to upstream docs for this migration, since you may have to make some other changes before upgrading your project config file's `layout` key.

camilamacedo86 reviewed (approved):
Just add a few nits. We might want to wait for v3 stable in the kb as well before gets it merged. 
However, /lgtm

estroz commented:
@camilamacedo86 the problem is `TestContext.CleanupManifests()` uses the global `kustomize` binary while go/v3 installs `kustomize` to `./bin/kustomize`. That function needs to be updated to consider a local `kustomize` path. In the meantime I've updated the cleanup step to use `make undeploy`, and added a comment about future refactoring.
